Bolts/Common/BFExecutor.m

```diff
@@ -10,6 +10,31 @@
 
 #import "BFExecutor.h"
 
+#import <pthread.h>
+
+/*!
+ Get the remaining stack-size of the current thread.
+
+ @param totalSize The total stack size of the current thread.
+
+ @return The remainging size, in bytes, available to the current thread.
+
+ @note This function cannot be inlined, as otherwise the internal implementation could fail to report the proper
+ remaining stack space.
+ */
+__attribute__((noinline)) static size_t remaining_stack_size(size_t *__nonnull restrict totalSize) {
+    pthread_t self = pthread_self();
```

Comment:
Nit: Maybe use something else instead of `self`?
It might be confusing, as self is not referring to instance/class, but to the thread?